### PR TITLE
feat: added mint functionality to PPO with relevant tests

### PR DESCRIPTION
**Pull request by Nilay27**
* Added method for minting tokens in PPO contract with relevant tests. 
* ~~By default, the `mint` method will always mint to the owner(i.e. the governance) of the contract~~
* Changed mint so that tokens can be minted to any address.

### CODE REVIEW

File: apps/smart-contracts/token/contracts/PPO.sol [hunk 1]
```diff
@@ -24,7 +24,9 @@ contract PPO is IPPO, ReentrancyGuardUpgradeable, SafeOwnableUpgradeable, ERC20B
     _transferHook = _newTransferHook;
   }
 
-  function mint(uint256 _amount) external override onlyOwner {}
+  function mint(uint256 _amount) external override onlyOwner {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/PPO.sol:27]: let's change this to be:
```suggestion
  function mint(address _recipient, uint256 _amount) external override onlyOwner {
```
for more flexibility (and if the owner is a timelock contract, we wouldn't want it minted to the timelock contract or it the PPO may end up stuck)

File: apps/smart-contracts/token/contracts/PPO.sol [hunk 2]
```diff
@@ -24,7 +24,9 @@ contract PPO is IPPO, ReentrancyGuardUpgradeable, SafeOwnableUpgradeable, ERC20B
...
+    _mint(owner(), _amount);
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/PPO.sol:28]:
```suggestion
    _mint(_recipient, _amount);
```

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 3]
```diff
@@ -91,4 +91,31 @@ describe('=> PPO', () => {
...
+
+    it("increases owner's balance", async () => {
+      const ownerPPOBalanceBefore = await ppo.balanceOf(owner.address)
+
+      await ppo.connect(owner).mint(1)
+
+      expect(await ppo.balanceOf(owner.address)).to.eq(ownerPPOBalanceBefore.add(1))
+    })
+
+    it("does not increase deployer's balance", async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/PPO.test.ts:113]: nit:
```suggestion
    it("doesn't increase deployer balance", async () => {
```

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 4]
```diff
@@ -91,4 +91,31 @@ describe('=> PPO', () => {
...
+    beforeEach(async () => {
+      await setupPPO()
+    })
+
+    it('reverts if not owner', async () => {
+      expect(await ppo.owner()).to.not.eq(user1.address)
+      await expect(ppo.connect(user1).mint(1)).revertedWith('Ownable: caller is not the owner')
+    })
+
+    it("increases owner's balance", async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/PPO.test.ts:105]: nit:
```suggestion
    it("increases owner balance", async () => {
```

File: apps/smart-contracts/token/contracts/PPO.sol (hunk 1 above)
Comment (XavierEkkel) [reply apps/smart-contracts/token/contracts/PPO.sol:27]: and add new tests for this (address zero + another address)

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 5]
```diff
@@ -91,4 +91,31 @@ describe('=> PPO', () => {
...
     })
   })
+
+  describe('# mint', () => {
+    beforeEach(async () => {
+      await setupPPO()
+    })
+
+    it('reverts if not owner', async () => {
+      expect(await ppo.owner()).to.not.eq(user1.address)
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/PPO.test.ts:101]:
```suggestion
      expect(await ppo.owner()).to.not.eq(user1.address)
      
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 6]
```diff
@@ -91,4 +91,31 @@ describe('=> PPO', () => {
       expect(await ppo.getTransferHook()).to.eq(JUNK_ADDRESS)
     })
   })
+
+  describe('# mint', () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/PPO.test.ts:95]: add tests for amount = 

- 0
- >1 (2)
- max uint

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 7]
```diff
@@ -91,4 +92,65 @@ describe('=> PPO', () => {
...
+
+    it('increases recipient balance if amount = 0', async () => {
+      const recipientPPOBalanceBefore = await ppo.balanceOf(owner.address)
+
+      await ppo.connect(owner).mint(user1.address, 0)
+
+      expect(await ppo.balanceOf(owner.address)).to.eq(recipientPPOBalanceBefore)
+    })
+
+    it('increases recipient balance if amount > 1', async () => {
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/test/PPO.test.ts:140]:
```suggestion
    it("mints amount > 1 to recipient", async () => {
```

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 8]
```diff
@@ -91,4 +92,65 @@ describe('=> PPO', () => {
...
+      const recipientPPOBalanceBefore = await ppo.balanceOf(owner.address)
+
+      await ppo.connect(owner).mint(user1.address, 2)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(recipientPPOBalanceBefore.add(2))
+    })
+
+    it('increases recipient balance if amount = max uint', async () => {
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/test/PPO.test.ts:148]:
```suggestion
    it('mints max uint to recipient', async () => {
```

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides test suite nits.

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 9]
```diff
@@ -3,6 +3,7 @@ import { ethers } from 'hardhat'
 import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address'
 import { ZERO_ADDRESS, JUNK_ADDRESS } from 'prepo-constants'
 import { ppoFixture } from './fixtures/PPOFixtures'
+import { MAX_UINT256 } from '../utils'
 import { PPO } from '../types/generated'
 
 describe('=> PPO', () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/PPO.test.ts:9]: add tests under initial state that balance and total supply are 0

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 10]
```diff
@@ -91,4 +92,65 @@ describe('=> PPO', () => {
...
+      )
+    })
+
+    it('reverts if minting to zero address', async () => {
+      await expect(ppo.connect(owner).mint(ZERO_ADDRESS, 1)).revertedWith(
+        'ERC20: mint to the zero address'
+      )
+    })
+
+    it('increases non-caller balance', async () => {
```
[the remark this answers is not in the document]
Comment (XavierEkkel) [reply apps/smart-contracts/token/test/PPO.test.ts:115]: nit:
```suggestion
    it('mints to non-caller if recipient is non-caller', async () => {
```
imo easier to understand if the main subject (recipient) mentioned first

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 11]
```diff
@@ -91,4 +92,65 @@ describe('=> PPO', () => {
...
+      const nonCallerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+      expect(owner).to.not.eq(user1)
+
+      await ppo.connect(owner).mint(user1.address, 1)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(nonCallerPPOBalanceBefore.add(1))
+    })
+
+    it('increases caller balance', async () => {
```
[the remark this answers is not in the document]
Comment (XavierEkkel) [reply apps/smart-contracts/token/test/PPO.test.ts:124]:
```suggestion
    it('mints to caller if recipient is caller', async () => {
```

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 12]
```diff
@@ -91,4 +92,65 @@ describe('=> PPO', () => {
...
+      const callerPPOBalanceBefore = await ppo.balanceOf(owner.address)
+
+      await ppo.connect(owner).mint(owner.address, 1)
+
+      expect(await ppo.balanceOf(owner.address)).to.eq(callerPPOBalanceBefore.add(1))
+    })
+
+    it('increases recipient balance if amount = 0', async () => {
```
[the remark this answers is not in the document]
Comment (XavierEkkel) [reply apps/smart-contracts/token/test/PPO.test.ts:132]: nit: don't need the apostrophe for recipient as it's perfectly understandable without it
```suggestion
    it('doesn't increase recipient balance if amount = 0', async () => {
```

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 13]
```diff
@@ -91,4 +92,65 @@ describe('=> PPO', () => {
       expect(await ppo.getTransferHook()).to.eq(JUNK_ADDRESS)
     })
   })
+
+  describe('# mint', () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/PPO.test.ts:96]: add test that it emits an event correctly in the 0 case and non-0 non-caller-recipient case
